### PR TITLE
Old account signin issue

### PR DESCRIPTION
**Pull request by Bhupesh-mfsi**

### CODE REVIEW

File: packages/lib/server-side/src/auth.ts [hunk 1]
```diff
@@ -96,8 +96,13 @@ export const Auth = (getAccount, serverSecret: string) => {
 
             res.json(jwt);
         } catch (e) {
-            next(e);
             console.error('unable to create new session token ', e);
+            if (e instanceof Error && e.message === 'Signature invalid') {
```
Comment (AlexNi245) [inline packages/lib/server-side/src/auth.ts:100]: i think we could return every error with code 400

     return res.status(400).json({
                    error: e.message,
                });

File: packages/messenger-widget/src/hooks/server-side/ServerSideConnector.ts [hunk 2]
```diff
@@ -100,11 +103,21 @@ export abstract class ServerSideConnector extends JwtInterceptor {
         );
 
         //Todo move to lib
-        const { data: newToken } = await axios.post(url, {
+        const {
```
Comment (AlexNi245) [inline packages/messenger-widget/src/hooks/server-side/ServerSideConnector.ts:106]: this might not work without catching the error

**Review by AlexNi245 — CHANGES_REQUESTED** [review]
see comments

File: packages/messenger-widget/src/hooks/server-side/ServerSideConnector.ts [hunk 3]
```diff
@@ -89,23 +93,39 @@ export abstract class ServerSideConnector extends JwtInterceptor {
...
+                data: newToken,
+                status,
+                data: error,
+            } = await axios.post(url, {
+                signature,
+                challenge,
+            });
 
-        return newToken;
+            return newToken;
```
Comment (AlexNi245) [inline packages/messenger-widget/src/hooks/server-side/ServerSideConnector.ts:119]: please add some comments why this step might occur

**Review by AlexNi245 — APPROVED** [review]

File: packages/messenger-widget/src/hooks/server-side/ServerSideConnector.ts (hunk 3 above)
Comment (Bhupesh-mfsi) [reply packages/messenger-widget/src/hooks/server-side/ServerSideConnector.ts:115]: added

**Pull request merged**